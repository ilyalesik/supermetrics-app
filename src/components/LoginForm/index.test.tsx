import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { LoginForm } from "./index";

it("renders LoginForm", () => {
  const tree = renderer.create(<LoginForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders LoginForm error", () => {
  const tree = renderer
    .create(<LoginForm error={new Error("Error")} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders LoginForm isLoading", () => {
  const tree = renderer.create(<LoginForm isLoading />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("LoginForm onSubmit works", () => {
  const onSubmit = jest.fn();
  render(<LoginForm onSubmit={onSubmit} />);
  fireEvent.input(screen.getByLabelText("name-input"), {
    target: { value: "John Doe" },
  });
  fireEvent.input(screen.getByLabelText("email-input"), {
    target: { value: "johndoe@example.com" },
  });
  fireEvent.click(screen.getByLabelText("submit-button"));
  expect(onSubmit.mock.calls.length).toBe(1);
  expect(onSubmit.mock.calls[0][0]).toMatchObject({
    name: "John Doe",
    email: "johndoe@example.com",
  });
});
