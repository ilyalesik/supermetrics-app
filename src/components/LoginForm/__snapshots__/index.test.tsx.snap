// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoginForm 1`] = `
<form
  className="login-form-container"
  onSubmit={[Function]}
>
  <div
    className="login-form-header"
  >
    <p>
      Login
    </p>
  </div>
  <div>
    <label
      className="login-form-label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      aria-label="name-input"
      className="login-form-input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="login-form-label"
      htmlFor="name"
    >
      Email
    </label>
    <input
      aria-label="email-input"
      className="login-form-input"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="login-form-footer"
  >
    <button
      aria-label="submit-button"
      className="login-form-button"
      type="submit"
    >
      Go
    </button>
  </div>
</form>
`;

exports[`renders LoginForm error 1`] = `
<form
  className="login-form-container"
  onSubmit={[Function]}
>
  <div
    className="login-form-header"
  >
    <p>
      Login
    </p>
  </div>
  <div>
    <label
      className="login-form-label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      aria-label="name-input"
      className="login-form-input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="login-form-label"
      htmlFor="name"
    >
      Email
    </label>
    <input
      aria-label="email-input"
      className="login-form-input"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div>
    <p
      className="login-form-error-message"
    >
      Error: 
      Error
    </p>
  </div>
  <div
    className="login-form-footer"
  >
    <button
      aria-label="submit-button"
      className="login-form-button"
      type="submit"
    >
      Go
    </button>
  </div>
</form>
`;

exports[`renders LoginForm isLoading 1`] = `
<form
  className="login-form-container"
  onSubmit={[Function]}
>
  <div
    className="login-form-header"
  >
    <p>
      Login
    </p>
  </div>
  <div>
    <label
      className="login-form-label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      aria-label="name-input"
      className="login-form-input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div>
    <label
      className="login-form-label"
      htmlFor="name"
    >
      Email
    </label>
    <input
      aria-label="email-input"
      className="login-form-input"
      onChange={[Function]}
      type="email"
      value=""
    />
  </div>
  <div
    className="login-form-footer"
  >
    <button
      aria-label="submit-button"
      className="login-form-button"
      disabled={true}
      type="submit"
    >
      Go
    </button>
  </div>
</form>
`;
